File: go/src/capturing_logger.go
package src

type CapturingLogger struct {
	lines []string
}

func (c *CapturingLogger) info(message string) {
	c.lines = append(c.lines, "INFO: "+message)
}
func (c *CapturingLogger) error(message string) {
	c.lines = append(c.lines, "ERROR: "+message)
}
func (c *CapturingLogger) getLoggedLines() []string {
	return c.lines
}

File: python/capturing_logger.py
class CapturingLogger:
    def __init__(self):
        self.lines = []

    def info(self, message):
        self.lines.append("INFO: " + message)

    def error(self, message):
        self.lines.append("ERROR: " + message)

    def get_logs(self):
        return self.lines
